modules/fedimint-wallet-client/src/deposit.rs

```diff
@@ -274,7 +284,11 @@ pub(crate) async fn transition_btc_tx_confirmed(
 
     let amount = pegin_proof.tx_output().value.into();
 
-    let wallet_input = WalletInput::new_v1(&pegin_proof);
+    let wallet_input = if consensus_version >= ModuleConsensusVersion::new(2, 2) {
+        WalletInput::new_v1(&pegin_proof)
+    } else {
+        WalletInput::new_v0(pegin_proof)
+    };
```

Comment:
Once we decommission the v0 input type in the client the fn signature should change such that it doesn't require a pegin proof anymore, but only the necessary data.